### PR TITLE
Check that num_nodes is both defined and not None, not just defined

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes https://github.com/kubernetes/kubernetes/issues/19318

@justinsb @zmerlynn

### CODE REVIEW

**Comment by justinsb** [comment]
Arrgghhh salt.  LGTM.

One day, I would love someone to explain `is defined` vs `get('num_nodes', None)` vs `get('num_nodes', '')` vs `['num_nodes']` to me :-)  And I'm sure I've missed a few!

**Comment by brendandburns** [comment]
@justinsb to be fair, this is Jinja, not Salt...

**Comment by zmerlynn** [comment]
Peeling back the problem, is #16071 even legit? The number of nodes in the cluster is relevant to the birth of the cluster but not the constant state of the cluster, which makes this solution limited. Is there a bug open to fix it for the MIG scaling case?

File: cluster/addons/cluster-monitoring/google/heapster-controller.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% set heapster_memory = "300Mi" -%}
-{% if pillar['num_nodes'] is defined -%}
+{% if pillar['num_nodes'] is defined and pillar['num_nodes'] is not None -%}
```
Comment (zmerlynn) [inline cluster/addons/cluster-monitoring/google/heapster-controller.yaml]: You could just shorthand this to `pillar.get('num_nodes') is not None`, couldn't you?

**Comment by brendandburns** [comment]
yeah, understood that it doesn't handle MIG scaling.  Doing so is _way_ more complicated...

**Comment by justinsb** [comment]
@k8s-bot test this

**Pull request merged**